src/approaches/interactive_learning_approach.py

```diff
@@ -117,7 +121,13 @@ def _relearn_predicates_and_nsrts(
                 (self._predicates_to_learn - {pred}) | {new_pred}
 
         # Learn NSRTs via superclass
-        self._learn_nsrts(dataset.trajectories, online_learning_cycle)
+        self._learn_nsrts(self._dataset.trajectories, online_learning_cycle)
+
+        # Save the things we need other than the NSRTs, which were already
```

Comment:
I think we need one other thing too -- `self._best_score`. and in general, we'll need to save/load any class attributes that can change after initialization. (cc @amburger66 )

Reply:
actually, best_score might get set correctly through the re-running of the interactions? but regardless, I think we should add it to the things we're saving and loading.